ts/twitter/src/app/(protected)/home/_components/timeline/timeline-post-card-popup-menu.tsx

```diff
@@ -0,0 +1,77 @@
+"use client";
+
+import { useColorModeValue } from "@/lib/components/ui/color-mode";
+import { Tooltip } from "@/lib/components/ui/tooltip";
+import { Flex, IconButton, Menu, Portal, Text } from "@chakra-ui/react";
+import {
+  AnalyticsIcon,
+  DeleteIcon,
+  EmbeddingIcon,
+  HighlightIcon,
+  ListIcon,
+  PinIcon,
+  ReplyIcon,
+  RequestIcon,
+  ThreeDotsIcon,
+} from "../../../../../lib/components/icons";
+
+const menuItems = [
+  { value: "delete", icon: DeleteIcon, text: "Delete", color: "red.400" },
+  { value: "pin", icon: PinIcon, text: "Pin to your profile" },
+  {
+    value: "highlight",
+    icon: HighlightIcon,
+    text: "Highlight on your profile",
+  },
+  { value: "list", icon: ListIcon, text: "Add/remove from Lists" },
+  { value: "reply", icon: ReplyIcon, text: "Change who can reply" },
+  { value: "engagements", icon: AnalyticsIcon, text: "View post engagements" },
+  { value: "embedding", icon: EmbeddingIcon, text: "Embed post" },
+  { value: "analytics", icon: AnalyticsIcon, text: "View post analytics" },
+  { value: "request", icon: RequestIcon, text: "Request community note" },
+];
+
+export const TimelinePostMenu: React.FC<Omit<Menu.RootProps, "children">> = (
+  props,
+) => {
+  return (
+    <Menu.Root {...props}>
+      <Menu.Trigger>
```

Comment:
There's a hydration error caused by nested `<button>` elements. The `Menu.Trigger` component renders a `<button>`, and the `IconButton` inside it also renders a `<button>`, which creates invalid HTML structure. Could you add the `asChild` prop to `Menu.Trigger` to fix this issue?
```suggestion
      <Menu.Trigger asChild>
```